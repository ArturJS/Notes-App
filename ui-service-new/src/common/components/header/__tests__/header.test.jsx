jest.mock('react-collapse', () => ({
    // eslint-disable-next-line react/prop-types
    Collapse: ({ children, ...props }) => (
        <div className="react-collapse">
            <div className="props">{JSON.stringify(props, null, '    ')}</div>
            <div className="children">{children}</div>
        </div>
    )
}));

/* eslint-disable import/first */
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Header, { HeaderClass } from '../header.jsx';
/* eslint-enable import/first */

let originalWindowScroll;
const mockWindowScroll = () => {
    originalWindowScroll = window.scroll;
    window.scroll = jest.fn();
};
const unmockWindowScroll = () => {
    window.scroll = originalWindowScroll;
};

describe('<FieldError />', () => {
    beforeEach(mockWindowScroll);
    afterEach(unmockWindowScroll);

    it('renders correctly', () => {
        const tree = renderer.create(<Header>test content</Header>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('toggle `isExpanded` state', () => {
        it('changes after click on .header-hamburger', () => {
            const wrapper = mount(<Header />);
            const { isExpanded: originalIsExpanded } = wrapper
                .find(HeaderClass)
                .props();

            wrapper.find('.header-hamburger').simulate('click');

            const { isExpanded } = wrapper.find(HeaderClass).props();

            expect(isExpanded).not.toBe(originalIsExpanded);
        });

        it('changes after keyPress on .header-hamburger', () => {
            const wrapper = mount(<Header />);
            const { isExpanded: originalIsExpanded } = wrapper
                .find(HeaderClass)
                .props();

            wrapper.find('.header-hamburger').simulate('keyPress');

            const { isExpanded } = wrapper.find(HeaderClass).props();

            expect(isExpanded).not.toBe(originalIsExpanded);
        });
    });

    describe('scrollTop', () => {
        it('scrolls to top after click on .icon-arrow-up', () => {
            const wrapper = mount(<Header />);

            wrapper.find('.icon-arrow-up').simulate('click');

            expect(window.scroll).toHaveBeenCalledWith({
                top: 0,
                behavior: 'smooth'
            });
        });

        it('scrolls to top after keyPress on .icon-arrow-up', () => {
            const wrapper = mount(<Header />);

            wrapper.find('.icon-arrow-up').simulate('keyPress');

            expect(window.scroll).toHaveBeenCalledWith({
                top: 0,
                behavior: 'smooth'
            });
        });
    });

    describe('scrollBottom', () => {
        it('scrolls to bottom after click on .icon-arrow-down', () => {
            const wrapper = mount(<Header />);

            wrapper.find('.icon-arrow-down').simulate('click');

            expect(window.scroll).toHaveBeenCalledWith({
                top: document.body.scrollHeight,
                behavior: 'smooth'
            });
        });

        it('scrolls to top after keyPress on .icon-arrow-up', () => {
            const wrapper = mount(<Header />);

            wrapper.find('.icon-arrow-down').simulate('keyPress');

            expect(window.scroll).toHaveBeenCalledWith({
                top: document.body.scrollHeight,
                behavior: 'smooth'
            });
        });
    });
});
