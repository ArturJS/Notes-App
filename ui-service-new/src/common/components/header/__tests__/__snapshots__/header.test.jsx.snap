// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldError /> renders correctly 1`] = `
<div
  className="header"
>
  <div
    className="header-content-wrapper"
  >
    <div
      className="header-content"
    >
      <i
        className="header-icon icon-arrow-up"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      />
      <i
        className="header-icon icon-arrow-down"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      />
      <div
        className="header-title"
      >
        Notes app
      </div>
      <div
        className="hide-from-tablet header-hamburger"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        <div
          className="header-hamburger-times"
        />
        <div
          className="header-hamburger-line"
        />
      </div>
    </div>
    <div
      className="react-collapse"
    >
      <div
        className="props"
      >
        {
    "className": "header-menu",
    "isOpened": false
}
      </div>
      <div
        className="children"
      >
        <div
          className="header-menu-content"
        >
          test content
        </div>
      </div>
    </div>
  </div>
</div>
`;
